src/context/Result.h

```diff
@@ -58,6 +66,20 @@ class Result final {
     }
 
     struct Core {
+        Core() = default;
+        Core(Core &&) = default;
+        Core& operator=(Core &&) = default;
+        Core(const Core &c) {
+            *this = c;
+        }
+        Core& operator=(const Core &c) {
```

Comment:
check `&c == this` to avoid self assignment.

Reply:
Fixed.